modules/cudf/src/series.ts

```diff
@@ -526,11 +527,31 @@ export {
   StructSeries,
 };
 
-function asColumn<T extends DataType>(value: SeriesProps<T>|Column<T>|arrow.Vector<T>): Column<T> {
+function inferType(value: any[]): DataType {
+  if (value.length == 0) return new Float64;
+  if (value.every((val) => typeof val === 'string')) return new Utf8String;
+  if (value.every((val) => typeof val === 'number' || val === null)) return new Float64;
+  if (value.every((val) => typeof val === 'bigint')) return new Int64;
```

Comment:
Could these checks be speculative? I.e. could we just check the first element and set the type based on that? If the subsequent series construction are guaranteed to raise a loud error in case the input is actually bad, then I think that might be acceptable, and spare the cost of these checks in the common case where things work.

Reply:
Interesting idea. Implementing this doesn't result in loud error later, infact some internal typecasting is done based on the first elements type:

```bash
> const { Series } = require('@rapidsai/cudf');

>Series.new([1, "two", 3, null]);
Float64Series { _col: Column {} } [ 1, NaN, 3, 0 ]

> Series.new(["1", "two", 3, null]);
StringSeries { _col: Column {} } [ '1', 'two', '3', null ]

```

Reply:
Well if something else doesn't raise an error down the line I think the checks here are unavoidable